### PR TITLE
Refine apply key generation

### PR DESCRIPTION
**Pull request by chaokunyang**
## What do these changes do?
This PR add func key cache to avoid duplicate func serialization when generate apply key which will be costly when func capture some big objects 

## Related issue number

Fixes #xxxx

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

**Review by zhongchun — DISMISSED** [review]
LGTM.

File: mars/utils.py [hunk 1]
```diff
@@ -1617,6 +1619,21 @@ def __str__(self):
...
+        if token is None:
+            fields = get_func_token_values(func)
+            token = tokenize(*fields)
+            _func_token_cache[func] = token
+        return token
+    except TypeError:  # cannot create weak reference to func like 'numpy.ufunc'
+        return tokenize(*get_func_token_values(func))
+
+
 def get_func_token_values(func):
```
Comment (fyrestone) [inline mars/utils.py:1637]: The `get_func_token_values` is not used by other modules except the `mars/utils.py`, it's better to rename it to `_get_func_token_values` to make it private.

Comment (chaokunyang) [reply mars/utils.py]: Updated

**Review by fyrestone — APPROVED** [review]
LGTM

**Review by zhongchun — APPROVED** [review]
LGTM.

**Pull request merged**